sdccc/src/test/java/com/draeger/medical/sdccc/tests/util/MdibHistorianTest.java

```diff
@@ -579,6 +582,361 @@ void testMultipleContextStates() throws Exception {
         }
     }
 
+    /**
+     * Tests if Reports are filtered out by GetAllUniqueReports when
+     * - they have the same MdibVersion.
+     */
+    @Test
+    void testGetAllUniqueReportsGoodDuplicateFilter() throws JAXBException, IOException {
+        final var operator1 = mdibBuilder.buildOperatorContextState("opDescriptor1", "opState");
+        final var operator2 = mdibBuilder.buildOperatorContextState("opDescriptor2", "opState");
+
+        final var contextReportMdibVersion = BigInteger.ONE;
+        final var contextReportStateVersion = BigInteger.TWO;
+        final var contextReport = buildEpisodicContextReport(
+                MdibBuilder.DEFAULT_SEQUENCE_ID, contextReportMdibVersion, contextReportStateVersion);
+
+        final var firstMod =
+                (EpisodicContextReport) contextReport.getBody().getAny().get(0);
+        firstMod.getReportPart().get(0).getContextState().add(operator1);
+        firstMod.getReportPart().get(0).getContextState().add(operator2);
+
+        messageStorageUtil.addInboundSecureHttpMessage(
+                storage, buildMdibEnvelope(MdibBuilder.DEFAULT_SEQUENCE_ID, BigInteger.ZERO));
+
+        messageStorageUtil.addInboundSecureHttpMessage(storage, contextReport);
+        messageStorageUtil.addInboundSecureHttpMessage(storage, contextReport); // duplicate
+        messageStorageUtil.addInboundSecureHttpMessage(storage, contextReport); // duplicate
+        messageStorageUtil.addInboundSecureHttpMessage(storage, contextReport); // duplicate
+
+        final var mockObserver = mock(TestRunObserver.class);
+        final var historian = historianFactory.createMdibHistorian(storage, mockObserver);
+
+        try (final var history = historian.getAllUniqueReports(MdibBuilder.DEFAULT_SEQUENCE_ID, BigInteger.ZERO)) {
+            final var list = history.toList();
+            assertEquals(1, list.size());
+            assertEquals(
+                    ((EpisodicContextReport) contextReport.getBody().getAny().get(0)).getMdibVersion(),
+                    list.get(0).getMdibVersion());
+            assertEquals(
+                    ((EpisodicContextReport) contextReport.getBody().getAny().get(0))
+                            .getReportPart()
+                            .size(),
+                    ((org.somda.sdc.biceps.model.message.EpisodicContextReport) list.get(0))
+                            .getReportPart()
+                            .size());
+            for (int i = 0;
+                    i
```

Comment:
Please fix the formatting.

Reply:
Fortunately, introducing a variable for the expression does the trick.

Reply:
Please do the same for `testGetAllUniqueReportsGoodDoNotFilterWhenReportTypeDiffers`

Reply:
Done.